### PR TITLE
Fix false positives on `redundant_objc_attribute` rule when using nested types

### LINKED ISSUE #2539: redundant_objc_attribute incorrectly flagging inner @objc enum
The following code triggers `redundant_objc_attribute`:

```swift
@objcMembers 
class Foo {
  @objc enum Bar { }
}
```

The code to flag a redundant `@objc` should only trigger if the declaration is a member, not an inner type.

I'm gonna take a stab at fixing this myself.

### PR DESCRIPTION
**Pull request by marcelofabri**
Fixes #2539

### CODE REVIEW

File: Rules.md [hunk 1]
```diff
@@ -15400,12 +15426,12 @@ extension Foo {
 ```
 
 ```swift
-@objc
```
Comment (dirtydanee) [inline Rules.md]: It seems from the diff that this triggering example was removed. Could you add it back to make sure it is triggered?

Comment (marcelofabri) [reply Rules.md]: I moved it to be a non-triggering example because the declaration is private (similar to https://github.com/realm/SwiftLint/pull/2501)

Comment (dirtydanee) [reply Rules.md]: Ok, i see, i agree, its a good idea to have it in the non triggering too.
However it is redundant to mark any attribute in an already `@objc` marked `extension`, regardless what the `ACL` is.

Comment (marcelofabri) [reply Rules.md]: Added it back in 0c92b948a9ebe52e30b44371ebf9cff9c4de8832

File: Source/SwiftLintFramework/Rules/Idiomatic/RedundantObjcAttributeRule.swift [hunk 2]
```diff
@@ -1,12 +1,11 @@
-import Foundation
 import SourceKittenFramework
 
 private let kindsImplyingObjc: Set<SwiftDeclarationAttributeKind> =
     [.ibaction, .iboutlet, .ibinspectable, .gkinspectable, .ibdesignable, .nsManaged]
 
 private let privateACL: Set<SwiftDeclarationAttributeKind> = [.private, .fileprivate]
```
Comment (dirtydanee) [inline Source/SwiftLintFramework/Rules/Idiomatic/RedundantObjcAttributeRule.swift]: Due to your last change (`!acl.isPrivate`), this attribute is not needed any more.

**Comment by dirtydanee** [comment]
> @dirtydanee @JaviSoto could you take a look at this and see if it makes sense? I'm not super familiar with `@objcMembers` to be 100% sure this is right.

I think the updated implementation is pretty neat, and its a much better approach to only check the parent structure instead of iterating on the whole file on each declaration kind.

**Comment by JaviSoto** [comment]
Hey! Sorry it took me so long to reply. Yeah this code looks so much better! It also solves the scenario I posted on #2539:

```swift
@objcMembers
class Foo {
  @objc enum Bar { }
}
```

But if I add it to the "non triggering examples", it's passing now 👍

**Comment by jpsim** [comment]
Nice work!

**Pull request merged**